### PR TITLE
[RTC-231] move server notifications to protobuffs

### PR DESCRIPTION
**Pull request by Rados13**

### CODE REVIEW

File: test/jellyfish_web/integration/server_socket_test.exs [hunk 1]
```diff
@@ -107,41 +106,47 @@ defmodule JellyfishWeb.Integration.ServerSocketTest do
...
+        case ControlMessage.decode(msg) do
+          %ControlMessage{
+            content: {:peerDisconnected, %PeerNotification{peerId: ^peer_id, roomId: ^room_id}}
+          } ->
+            assert true
+
+          _other ->
+            nil
+        end
+    end
```
Comment (sgfn) [inline test/jellyfish_web/integration/server_socket_test.exs:129]: Is there a specific reason why this `receive` block is different than the one from lines 85-90?

**Review by sgfn — CHANGES_REQUESTED** [review]

File: .formatter.exs [hunk 2]
```diff
@@ -1,4 +1,8 @@
 [
   import_deps: [:phoenix, :open_api_spex],
-  inputs: ["*.{ex,exs}", "{config,lib,examples,test}/**/*.{ex,exs}"]
+  inputs:
```
Comment (mickel8) [inline .formatter.exs:3]: As in the server SDK, we can use `import_deps: [:protobuf]`

File: compile_proto.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
Comment (mickel8) [inline compile_proto.sh:1]: Just remember to adjust it after we decide the final form in the elixir server SDK

File: test/jellyfish_web/integration/server_socket_test.exs [hunk 4]
```diff
@@ -85,10 +82,12 @@ defmodule JellyfishWeb.Integration.ServerSocketTest do
...
-    assert_receive %{
-      "data" => %{"roomId" => ^room_id, "type" => "roomCrashed"},
-      "type" => "controlMessage"
-    }
+    receive do
+      msg when is_binary(msg) ->
+        assert %ControlMessage{
+                 content: {:roomCrashed, %RoomNotification{roomId: ^room_id}}
+               } = ControlMessage.decode(msg)
+    end
```
Comment (mickel8) [inline test/jellyfish_web/integration/server_socket_test.exs:90]: Let's decode message in the ws.ex and assert_receive against ControlMessage structs

File: test/jellyfish_web/integration/server_socket_test.exs [hunk 5]
```diff
@@ -107,41 +106,47 @@ defmodule JellyfishWeb.Integration.ServerSocketTest do
...
-    auth_request = auth_request(peer_token)
+    auth_request = peer_auth_request(peer_token)
 
     :ok = WS.send_frame(peer_ws, auth_request)
 
-    assert_receive %{
-      "data" => %{"id" => ^peer_id, "type" => "peerConnected"},
-      "type" => "controlMessage"
-    }
+    assert_receive %{"data" => %{"type" => "authenticated"}, "type" => "controlMessage"}
```
Comment (mickel8) [inline test/jellyfish_web/integration/server_socket_test.exs:113]: IMO, this isn't needed as we test peer auth in peer_socket tests

File: test/jellyfish_web/integration/server_socket_test.exs [hunk 6]
```diff
@@ -107,41 +106,47 @@ defmodule JellyfishWeb.Integration.ServerSocketTest do
...
 
-    assert_receive %{
-      "data" => %{"id" => ^peer_id, "type" => "peerDisconnected"},
-      "type" => "controlMessage"
-    }
+    receive do
+      msg when is_binary(msg) ->
+        case ControlMessage.decode(msg) do
+          %ControlMessage{
+            content: {:peerDisconnected, %PeerNotification{peerId: ^peer_id, roomId: ^room_id}}
```
Comment (mickel8) [inline test/jellyfish_web/integration/server_socket_test.exs:122]: Checking if we got peerConnected is also important. In particular, the fact that we got peerDisconnected doesn't mean we got peerConnected

**Review by mickel8 — CHANGES_REQUESTED** [review]

**Review by sgfn — APPROVED** [review]

File: compile_proto.sh [hunk 7]
```diff
@@ -3,7 +3,7 @@
 # Terminate on errors
 set -e
 
-deps="printf find git protoc"
+deps="printf find git protoc protoc-gen-elixir echo"
 
 for dep in $deps; do
   if ! [[ $(which $dep) ]]; then
```
Comment (mickel8) [inline compile_proto.sh]: As in the server SDK, I don't think we need to check presence of those commands

File: test/support/ws.ex [hunk 8]
```diff
@@ -21,7 +22,7 @@ defmodule JellyfishWeb.WS do
 
   @impl true
   def handle_frame({:binary, msg}, state) do
-    send(state.caller, msg)
+    send(state.caller, ControlMessage.decode(msg))
```
Comment (mickel8) [inline test/support/ws.ex:25]: that's a nitpick but I would unpack outer ControlMessage here. In actual tests we will be able to do 

```elixir
    assert_receive  %PeerDisconnected{peer_id: ^peer_id, room_id: ^room_id}
```

File: .gitmodules [hunk 9]
```diff
@@ -0,0 +1,4 @@
+[submodule "protos"]
+	path = protos
+	url = https://github.com/jellyfish-dev/protos.git
+    branch = master
```
Comment (LVala) [inline .gitmodules:4]: This uneven indentation bothers me a bit.

**Review by LVala — APPROVED** [review]
👍

**Pull request merged**